pkg/retry/retry.go

```diff
@@ -59,6 +61,9 @@ func isRetryable(err error) bool {
 	case *net.OpError:
 		return isRetryable(e.Err)
 	case *url.Error:
+		if strings.HasSuffix(e.Error(), io.EOF.Error()) {
+			return true
+		}
 		return isRetryable(e.Err)
```

Comment:
Couldn’t this be done without the string matching? At the very least check `e.Err` instead of all of `e`. Ideally just an explicit check for `io.EOF`, although I suppose that would require reproducing the error and verifying that there isn’t an intermediate value.

Reply:
makes sense, the e.Error() just appends the e.Err at the end. 
I haven't found a way to reproduce the EOF error, just tried with the tcp i/o timeout..

Reply:
Thanks.